x-pack/plugins/apm/public/components/alerting/utils/fields.tsx

```diff
@@ -98,6 +99,46 @@ export function EnvironmentField({
   );
 }
 
+export function TransactionNameField({
+  currentValue,
+  onChange,
+  serviceName,
+}: {
+  currentValue?: string;
+  onChange: (value?: string) => void;
+  serviceName?: string;
+}) {
+  const label = i18n.translate('xpack.apm.alerting.fields.transaction.name', {
+    defaultMessage: 'Name',
+  });
+
+  return (
+    <PopoverExpression value={currentValue || allOptionText} title={label}>
+      <SuggestionsSelect
```

Comment:
We may want to sort the suggestions. Currently they are ordered by volume (afaict)

<img width="306" alt="image" src="https://user-images.githubusercontent.com/209966/233335868-42197803-8e0a-40f4-8d62-95b556a82b75.png">

Reply:
@sqren do you mean sort alphabetically? The suggestions dropdown uses _terms_enum API which I'm not 100% how it returns the results

Reply:
Yes, I mean alphabetically. Sorting on the client is fine.